go.mod

```diff
@@ -35,6 +35,7 @@ require (
 	google.golang.org/grpc v1.42.0
 	google.golang.org/protobuf v1.27.1
 	gopkg.in/yaml.v2 v2.4.0
+	istio.io/client-go v1.13.3
```

Comment:
I don't think we should be adding this dependency to the project. Traffic-routers were implemented with dynamic go-client approach to avoid having to add/maintain this type of dependency.

Reply:
I thought about that as well, however I went ahead with adding it because it allows us to validate the referenced type locally. If we were to remove the dependency we would just be able to check that the small pieces we use are correct and we won't validate the correctness as far as schema or extra fields are concerned on referenced resources which is maybe ok I guess. I thought it was a nice though to run lint and also get some basic checks that your VirtualService for is valid as far as schema and no extra fields are concerned.

Reply:
I can for sure re work it though and drop that validation since it would require us adding a lot of dependencies.